src/librustc_mir/borrow_check/nll/invalidation.rs

```diff
@@ -0,0 +1,955 @@
+                values: _,
+                targets: _,
+            } => {
+                self.consume_operand(ContextKind::SwitchInt.new(location), discr);
+            }
+            TerminatorKind::Drop {
+                location: ref drop_place,
+                target: _,
+                unwind: _,
+            } => {
+                let tcx = self.infcx.tcx;
+                let gcx = tcx.global_tcx();
+                let drop_place_ty = drop_place.ty(self.mir, tcx);
+                let drop_place_ty = tcx.erase_regions(&drop_place_ty).to_ty(tcx);
+                let drop_place_ty = gcx.lift(&drop_place_ty).unwrap();
+                self.visit_terminator_drop(location, terminator, drop_place, drop_place_ty);
+            }
+            TerminatorKind::DropAndReplace {
+                location: ref drop_place,
+                value: ref new_value,
+                target: _,
+                unwind: _,
+            } => {
+                self.mutate_place(
+                    ContextKind::DropAndReplace.new(location),
+                    drop_place,
+                    Deep,
+                    JustWrite,
+                );
+                self.consume_operand(
+                    ContextKind::DropAndReplace.new(location),
+                    new_value,
+                );
+            }
+            TerminatorKind::Call {
+                ref func,
+                ref args,
+                ref destination,
+                cleanup: _,
+            } => {
+                self.consume_operand(ContextKind::CallOperator.new(location), func);
+                for arg in args {
+                    self.consume_operand(ContextKind::CallOperand.new(location), arg);
+                }
+                if let Some((ref dest, _ /*bb*/)) = *destination {
+                    self.mutate_place(
+                        ContextKind::CallDest.new(location),
+                        dest,
+                        Deep,
+                        JustWrite,
+                    );
+                }
+            }
+            TerminatorKind::Assert {
+                ref cond,
+                expected: _,
+                ref msg,
+                target: _,
+                cleanup: _,
+            } => {
+                self.consume_operand(ContextKind::Assert.new(location), cond);
+                use rustc::mir::interpret::EvalErrorKind::BoundsCheck;
+                if let BoundsCheck { ref len, ref index } = *msg {
+                    self.consume_operand(ContextKind::Assert.new(location), len);
+                    self.consume_operand(ContextKind::Assert.new(location), index);
+                }
+            }
+            TerminatorKind::Yield {
+                ref value,
+                resume: _,
+                drop: _,
+            } => {
+                self.consume_operand(ContextKind::Yield.new(location), value);
+
+                // ** FIXME(bob_twinkles) figure out what the equivalent of this is
```

Comment:
the equivalent here is invalidating the loans at the point *after* the yield. Basically we want to invalidate all loans that are borrowing "local" content (something owned by current stack frame) at this point:

```rust
let resume_index = self.location_table.start_index(resume.start_location())
```

(That is taken from [my branch that integrated with the new approach](https://github.com/nikomatsakis/rust/blob/5656960ea0b62624341a10c175ab9d154a5fe94c/src/librustc_mir/borrow_check/mod.rs#L587-L603))